SRS.MD

```diff
@@ -0,0 +1,308 @@
+
+# System Features
+
+## Database
+
+### Database - Description and Priority
+
+Description: Database to store data of rooms, customers and billing details.
+
+Priority: <span style="color: red;">**High**<span>
+
+### Database - Stimulus/Response Sequences
+
+Will function as directed by the Server program via the JDBC driver.
+
+### Database - Functional Requirements
+
+(Refer to ER Model in the SDD)
+
+## Server
+
+### Server - Description and Priority
+
+A Java-based server application to facilitate backend operations, including User Authentication and performing Database Operations.
+
+Priority: <span style="color: red;">**High**<span>
+
+### Server - Stimulus/Response Sequences
+
+The user can Start or Suspend the server.
+
+### Server - Functional Requirements
+
+- Connect to DB via JDBC
+- Authentication and Handling of User Login
+- Handle Client side requests via API
+- Add/Remove/Modify Users
+- Add/Remove/Modify Customers
+- Add/Remove/Modify Hotels
+- Add/Remove/Modify Rooms
+- Log Modifications with Timestamp on DB
+
+## Client Side GUI
+
+### GUI - Description and Priority
+
+A JavaFX based GUI Application to ensure usability and enable API calls to the server application.
+
+Priority: <span style="color: orange;">**Medium**<span>
+
+### GUI - Stimulus/Response Sequences
+
+(Refer to GUI SDD)
+
+### GUI - Functional Requirements
+
+- Authentication and Handling of User Login
+- Handle Client side requests via API
+- GUI Functions to List/Add/Remove/Modify Users/Customers/Hotels/Rooms
+- Log Connections, Messages and Operations with Timestamp Locally
+
+---
+
+<div style="page-break-after: always; visibility: hidden; color: #000;">\pagebreak</div>
+
+# External Interface Requirements
+
+## User Interfaces
+
+GUI shall be provided by the Client Side application, and the same must be usable via mouse/touchscreen and keyboard.
+
+## Hardware Interfaces
+
+Devices capable of Java Runtime Environment v17+ with keyboard, display and mouse.
+
+## Software Interfaces
+
+The following software(s) and/or libraries shall be used for the project:
+
+- _Java Platform with Maven:_ For compilation, building and running the projects.
+- _JavaFX:_ To enable GUI on Java Platform.
+- _PostgreSQL:_ For Database.
+- _Spring Boot Framework:_ For back-end server.
+- _PostgreSQL JDBC Driver:_ To connect the Java-based project to Database.
+
+## Communication Interfaces
+
+To enable server-client communication, network is necessary. The server and the client computers chould be connected to the same WLAN/LAN via HTTP(S) and/or required protocols.
+
+---
+
+<div style="page-break-after: always; visibility: hidden; color: #000;">\pagebreak</div>
+
+# Other Non-Functional Interfaces
+
+## Performance Requirements
+
+None spectified at the time of writing SRS.
+
+## Safety Requirements
+
+None spectified at the time of writing SRS.
+
+## Security Requirements
+
+Encryption of User credentials and customer details via Military Grade Encryption Algorithms.
```

Comment:
Encryption algorithm and hash algorithm to be written here.

Reply:
Added Encryption details.